### PR TITLE
Added bucket sort

### LINKED ISSUE #375: bucket sort
## 📚 Documentation

(A clear and concise description of what the issue is.)
bucket sort
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

(Write your answer here.)
yes

### PR DESCRIPTION
**Pull request by nagasrihith608**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
yes
(Write your answer here.)

### Description
Added bucket sort
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixed #375 
(Write your answer here.)

### CODE REVIEW

File: docs/sorting-algorithms/bucket-sort.mdx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    { label: 'C++', value: 'cpp', },
+    { label: 'Java', value: 'java', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+// Header file to include STL library
+#include<bits/stdc++.h>
+ //For cin and cout
```
Comment (vybhav72954) [inline docs/sorting-algorithms/bucket-sort.mdx:49]: Comment should look like this `// For cin and cout`

File: docs/sorting-algorithms/bucket-sort.mdx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+using namespace std;
+
+//to sort the individual bucket
```
Comment (vybhav72954) [inline docs/sorting-algorithms/bucket-sort.mdx:52]: Style the comment

File: docs/sorting-algorithms/bucket-sort.mdx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    temp[idx][count[idx]] = arr[i];
+    count[idx]++;
+  }
+
+  for (int i = 0; i < 10; i++) {
+    if (count[i] > 0)
+      Insertion_Sort(temp, count[i], i);
+  }
+
+  //to rewrite the values in array in sorted order
```
Comment (vybhav72954) [inline docs/sorting-algorithms/bucket-sort.mdx:84]: Style the comment

File: docs/sorting-algorithms/bucket-sort.mdx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  int n;
+  // Declare an array of float numbers
+  float arr[20];
+  cout << "Enter number of elements : ";
+  cin >> n;
+  for (int i = 0; i < n; i++) {
+    cout << "Enter the elements: ";
+    cin >> arr[i];
+  }
+  //loop to print original array
```
Comment (vybhav72954) [inline docs/sorting-algorithms/bucket-sort.mdx:107]: Style the comment

File: docs/sorting-algorithms/bucket-sort.mdx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+            int bucketIndex = (int) ((arr[i] * k) / max_val);
+            bucketList.get(bucketIndex).add(arr[i]);
+        }
+
+        // sorting each bucket one by one
+        for (int i=0; i<k; i++) {
+            Collections.sort(bucketList.get(i));
+        }
+
+        //joining buckets
```
Comment (vybhav72954) [inline docs/sorting-algorithms/bucket-sort.mdx:159]: Style the comment

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
Can you rebase against the master to fix the build?

**Comment by vybhav72954** [comment]
Updates?

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this due to inactivity. Thank you for your valuable time.😊

**Pull request closed without merging**